pkg/api/apply.go

```diff
@@ -1,11 +1,40 @@
 package api
 
 import (
+	"fmt"
+	"strings"
+
+	argo "github.com/argoproj/argo-cd/pkg/client/clientset/versioned"
 	"github.com/pkg/errors"
 	"github.com/rebuy-de/kubernetes-deployment/pkg/statsdw"
 	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
 )
 
+func checkForArgoApp(project string) (bool, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return true, errors.Wrap(err, "failed to load kubernetes config for Argo")
+	}
+
+	clientset, err := argo.NewForConfig(config)
+	if err != nil {
+		return true, errors.Wrap(err, "failed to initialize kubernetes client for Argo")
+	}
+
+	argoClient := clientset.ArgoprojV1alpha1()
+
+	appsClient := argoClient.Applications("")
+	_, err = appsClient.Get(project, v1.GetOptions{})
+	if err != nil {
+		log.Warnf("failed to get argo app '%s' - legacy project", project)
```

Comment:
Wouldn't that mean that we get this error message on every legacy deployment.

Reply:
I changed it to a debug message, I just want to make sure we notice any errors in the logs.

Reply:
In this case you should examine the error type and check if it is a not found error or something other.